src/python/pants/binaries/binary_tool.py

```diff
@@ -50,6 +51,10 @@ class BinaryToolBase(Subsystem):
   # Subclasses may set this to provide extra register() kwargs for the --version option.
   extra_version_option_kwargs = None
 
+  def __init__(self, *args, **kwargs):
+    super(BinaryToolBase, self).__init__(*args, **kwargs)
+    self._snapshot_lock = threading.RLock()
```

Comment:
Why an `RLock` not just a `Lock`? Re-entrance seems unlikely?

Reply:
Mm, yea, it can just be a `Lock`. Was copypasta.